generator/src/main/java/com/scottlogic/deg/generator/generation/FilterAfterFirst.java

```diff
@@ -0,0 +1,24 @@
+package com.scottlogic.deg.generator.generation;
+
+public class FilterAfterFirst<T> {
```

Comment:
This type is used in a filter to ensure that only one instance of a value is emitted. This is the same behaviour as `distinct`, can this be used instead to remove this type (and its extra complexity).

If not, please add a javadoc to explain what it does, and maybe consider changing its name to better describe its purpose rather than where it could be used, something like a `DistinctValueFilter` or `SingleValueGate` might better describe its behaviour.

Better still would be to refactor `StandardFieldValueSourceEvaluator` to ensure that `nullValueSource` is never added/projected more than once therefore removing the need for the `filter` or `distinct` entirely.

My preference would be to use `distinct` first, followed by a refactor of `StandardFieldValueSourceEvaluator` then if the former two cannot be achieved, then this type is renamed and documented.

Reply:
replaced `FilterAfterFirst` class with `.distinct()`